### ISSUE TITLE
Declare first script to load in a collection

### ISSUE THREAD

**Issue by jcbk101** [defold/defold#8581]
When Defold loads a collection with GOs and Guis, etc. There is no way to determine which script gets loaded first.

I think there should be a way to set a single script in a collection as the `main.script` file for that collection. No need for ordered scripts. All that is needed is to be able to set a single script as a collection's `main.script` file. 

Reason: For example, I have a map loader module the gets called from my `game.script` file, which is the main script for the collection called game. But that script is not called first, my player controller script is called first which relies on data from the game.script file. This causes bugs and timing issues. setting a delay works for the most part, but I cannot trust this to be consistent.

**Comment by swajj** [comment]
I once determined that top-level code in the render script is the very first code run- that's where I do critical initialization that can't wait for the main scrit's init() function. And any other script files I assume that init() hasn't run in their inits- if I care enough, I have one-shot code in that script's update() gated on a boolean "is_init" flag.

But all that said, it would be nice to have a more "officially supported" way to know certain code is the "very first code run" in some form or another.

**Comment by jcbk101** [comment]
@britzl Is there an indicator at least that Defold may use to call a script first? Maybe alphabetical order in file names?

**Comment by britzl** [comment]
> Is there an indicator at least that Defold may use to call a script first? Maybe alphabetical order in file names?

No, there are no such indicators and we don't want to promise a certain order of initialisation.

**Comment by JCash** [comment]
At the very top, you have the boot collection (see game.project properties).
From there you should be able to manage a strict chain of events.

E.g. when the boostrap script has been called, and the [collection proxy has loaded](https://defold.com/manuals/collection-proxy/#loading-a-collection) the map/level, then you enable the collection by sending events "init" and "enable".
After that, you know that all scripts are setup, you can then send a message to which ever script in that proxy to "start" the next set of events.

As we feel there are ways to control this in a satisfactory way, we will close this feature request for now.
